feat: target C#11 with safeguards for `CSharp.lua` compatibility

alrun3 opened the pull request:
We're using .NET 9, which defaults to C#13, but CSharp.lua only supports up to C#10. Some C#11 features (e.g., `required` properties) are already in use, but others are unsupported and can break transpilation.

To ensure compatibility and prevent invalid suggestions:

* Set `LangVersion` to 11.0 in all transpiled projects
* Add an `.editorconfig` file per transpiled project to suppress C# features not supported by CSharp.lua
* Suppress remaining informational diagnostics using `.editorconfig` and `#pragma warning disable` in generated files and vendored projects (i.e., source-copied external dependencies)

_No suppressions are added to maintained projects that fully support these features_

As a result, all informational messages have been resolved in the transpiled projects. Since the transpiled projects do not share a common directory, the `.editorconfig` settings are duplicated per project. If the directory structure is refactored in the future, these configurations could be consolidated to reduce duplication

Closes: #3440

YakaryBovine commented on src/WarcraftLegacies.Shared/Constants.cs:
```diff
@@ -1,3 +1,4 @@
+﻿#pragma warning disable
```
This will get removed when Constants.cs and Regions.cs are regenerated, which happens on every W3X-->MapData process. I think we'll need to prepend it on generation somehow?

YakaryBovine replied:
Cool, so sounds like we should
1. Prepend the pragma warning disable after generation, so this can be merged
2. Ask Arxos to make the same change upstream

YakaryBovine reviewed (approved):
Awesome, happy to take the patchwork Constants approach given how useful being able to just rely on IDE guidance is going to be.

alrun3 commented:
Just repeating this here: Nice to see it merged! However, just a heads-up: the `fixup!` commits aren't meant to be included in the main branch. They're meta commits intended to be amended into their target commits. I left them as-is to make it easier to track changes since the last review. Normally, you'd either squash merge the PR or use `--autosquash` during rebase to handle these
